services/ui-src/src/utils/reports/entities.ts

```diff
@@ -56,3 +64,70 @@ export const entityWasUpdated = (
   originalEntity: EntityShape,
   newEntity: AnyObject
 ) => JSON.stringify(originalEntity) !== JSON.stringify(newEntity);
+
+// rendering dynamic Target Population fields on the WP and SAR forms
+export const renderTargetPopulationFields = (
+  report: ReportShape | undefined,
+  fields: (FormField | FormLayoutElement)[],
+  location: AnyObject
+) => {
+  const updatedTargetPopulations =
+    report?.reportType === ReportType.SAR
+      ? report?.fieldData?.workPlanData.targetPopulations
```

Comment:
based on the type update wouldn't this be `report.workPlanData.fieldData.targetPopulations`? 

I'm just guessing here, but it doesn't look like workPlanData is nested under fieldData, right?